core/io/resource_importer.cpp

```diff
@@ -118,9 +118,22 @@ Error ResourceFormatImporter::_get_path_and_type(const String &p_path, PathAndTy
 	}
 #endif
 
-	if (r_path_and_type.path.is_empty() || r_path_and_type.type.is_empty()) {
+	if (r_path_and_type.type.is_empty()) {
 		return ERR_FILE_CORRUPT;
 	}
+	if (r_path_and_type.path.is_empty()) {
+		// The ResourceImporterCSVTranslation does not write files in the .godot folder
+		// it's normal if the path is empty.
+		if (r_path_and_type.importer.is_empty()) {
+			return ERR_FILE_CORRUPT;
+		}
+
+		// It's only invalid if the extension for the importer is not empty.
+		Ref<ResourceImporter> importer = get_importer_by_name(r_path_and_type.importer);
+		if (importer.is_null() || !importer->get_save_extension().is_empty()) {
```

Comment:
```suggestion
		if (importer.is_null() || importer->get_save_extension().is_empty()) {
```
If it's supposed to be non-empty...

Reply:
On the contrary, the ERR_FILE_CORRUPT should be raised only if the importer has an extension (not is_empty). If no extension, we want to return OK.

Reply:
I misread the comment.